src/Marten/Events/Daemon/Coordination/ProjectionCoordinator.cs

```diff
@@ -0,0 +1,252 @@
+    {
+    }
+}
+
+public class ProjectionCoordinator : IProjectionCoordinator
+{
+    private readonly StoreOptions _options;
+    private readonly ILogger<ProjectionCoordinator> _logger;
+
+    private readonly ConcurrentDictionary<IMartenDatabase, IProjectionDaemon> _daemons = new();
+    private readonly ResiliencePipeline _resilience;
+    private readonly CancellationTokenSource _cancellation = new();
+    private Task _runner;
+
+    public ProjectionCoordinator(IDocumentStore documentStore, ILogger<ProjectionCoordinator> logger)
+    {
+        var store = (DocumentStore)documentStore;
+
+        if (store.Options.Projections.AsyncMode == DaemonMode.Solo)
+        {
+            Distributor = new SoloProjectionDistributor(store);
+        }
+        else if (store.Options.Projections.AsyncMode == DaemonMode.HotCold)
+        {
+            if (store.Options.Tenancy is DefaultTenancy)
+            {
+                Distributor = new SingleTenantProjectionDistributor(store);
+            }
+            else
+            {
+                Distributor = new MultiTenantedProjectionDistributor(store);
+            }
+        }
+
+        _options = store.Options;
+        _logger = logger;
+        _resilience = store.Options.ResiliencePipeline;
+
+        Store = store;
+    }
+
+    public IProjectionDaemon DaemonForMainDatabase()
+    {
+        var database = (MartenDatabase)Store.Tenancy.Default.Database;
+        if (_daemons.TryGetValue(database, out var daemon))
+        {
+            return daemon;
+        }
+
+        daemon = database.StartProjectionDaemon(Store, _logger);
+        _daemons[database] = daemon;
+
+        return daemon;
+    }
+
+    public async ValueTask<IProjectionDaemon> DaemonForDatabase(string databaseIdentifier)
+    {
+        var database = (MartenDatabase)await Store.Storage.FindOrCreateDatabase(databaseIdentifier).ConfigureAwait(false);
+        if (_daemons.TryGetValue(database, out var daemon))
+        {
+            return daemon;
+        }
+
+        daemon = database.StartProjectionDaemon(Store, _logger);
+        _daemons[database] = daemon;
+
+        return daemon;
+    }
+
+    public DocumentStore Store { get; }
+
+    public IProjectionDistributor Distributor { get; }
+
+    internal record DaemonShardName(IProjectionDaemon Daemon, ShardName Name);
+
+    public Task StartAsync(CancellationToken cancellationToken)
+    {
+        _runner = Task.Run(() => executeAsync(_cancellation.Token), _cancellation.Token);
+
+        return Task.CompletedTask;
+    }
+
+    public async Task StopAsync(CancellationToken cancellationToken)
+    {
+#if NET8_0_OR_GREATER
+        await _cancellation.CancelAsync().ConfigureAwait(false);
+#else
+        _cancellation.Cancel();
+#endif
+
+        try
+        {
+#pragma warning disable VSTHRD003
+            await _runner.ConfigureAwait(false);
+#pragma warning restore VSTHRD003
+        }
+        catch (Exception)
+        {
```

Comment:
**CONCERN:** Shouldn't we add at least a log entry here?

Reply:
Did so. Not sure it's valuable, but why not just in case because I've been wrong about that before. Did have it ignore TaskCanceledException though.